### PR TITLE
Loadtestservice - SDK initial review

### PR DESCRIPTION
**Pull request by Yashikaj14**
SDK is being regenerated based on a new swagger spec and the upload file function has been written.

cmd:
```
autorest --version=3.7.2 --use=@autorest/python@5.16.0 --use=@autorest/modelerfour@4.19.3 README.md
```

### CODE REVIEW

File: sdk/loadtestservice/azure-analytics-loadtestservice/MANIFEST.in [hunk 1]
```diff
@@ -0,0 +1,6 @@
+include *.md
+include azure/__init__.py
+include azure/analytics/__init__.py
+include LICENSE
+recursive-include tests *.py
+recursive-include samples *.py *.md
```
Comment (msyyc) [inline sdk/loadtestservice/azure-analytics-loadtestservice/MANIFEST.in:6]:
```suggestion
recursive-include samples *.py *.md
include azure/analytics/loadtestservice/py.typed
```

File: sdk/loadtestservice/azure-analytics-loadtestservice/setup.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        "Programming Language :: Python :: 3 :: Only",
+        "Programming Language :: Python :: 3",
+        "Programming Language :: Python :: 3.6",
+        "Programming Language :: Python :: 3.7",
+        "Programming Language :: Python :: 3.8",
+        "Programming Language :: Python :: 3.9",
+        "Programming Language :: Python :: 3.10",
+        "License :: OSI Approved :: MIT License",
+    ],
+    zip_safe=False,
```
Comment (msyyc) [inline sdk/loadtestservice/azure-analytics-loadtestservice/setup.py:50]:
```suggestion
    include_package_data=True,
    package_data={
        "pytyped": ["py.typed"],
    },
    zip_safe=False,
```

**Comment by msyyc** [comment]
Hi @Yashikaj14 DPG tool generates framework for SDK and there are some files (README.md, test, samples) that need to be customized by you for better usage experience of customers.

**Comment by msyyc** [comment]
Hi @Yashikaj14 The test actually does not run. Please follow https://github.com/Azure/azure-sdk-for-python/wiki/Dataplane-Codegen-Quick-Start-for-Test to update test. You could also refer to test of other service: https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/purview/azure-purview-administration/tests/test_smoke.py

File: sdk/loadtestservice/azure-analytics-loadtestservice/README.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+## Examples
+
+```python
+>>> from azure.analytics.loadtestservice import LoadTestClient
+>>> from azure.identity import DefaultAzureCredential
+>>> from azure.core.exceptions import HttpResponseError
+
+>>> client = LoadTestClient(endpoint='<endpoint>', credential=DefaultAzureCredential())
+>>> try:
+        <!-- write test code here -->
```
Comment (msyyc) [inline sdk/loadtestservice/azure-analytics-loadtestservice/README.md:53]: please update here. The `Examples` part shall show customers how to use the package quickly

File: sdk/loadtestservice/azure-analytics-loadtestservice/samples/sample.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+# Build a client through AAD
+client = LoadTestClient(credential=DefaultAzureCredential(), endpoint=endpoint)
+
+# write your sample here. For example:
+# try:
+#     result = client.xxx.xx(...)
+#     print(result)
+# except HttpResponseError as e:
+#     print('Failed to send JSON message: {}'.format(e.response.json()))
```
Comment (msyyc) [inline sdk/loadtestservice/azure-analytics-loadtestservice/samples/sample.py:53]: please update the code. The sample could show the hero scenario that you want to customers to know

File: sdk/loadtestservice/azure-analytics-loadtestservice/swagger/README.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
+### Settings
+
+```yaml
+input-file: D:\Project\azure-rest-api-specs\specification\loadtestservice\data-plane\Microsoft.LoadTestService\preview\2022-06-01-preview\loadtestservice.json
```
Comment (msyyc) [inline sdk/loadtestservice/azure-analytics-loadtestservice/swagger/README.md:4]: the `input-file` shall be http link like : https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/purview/azure-purview-catalog/swagger#settings

File: sdk/loadtestservice/azure-analytics-loadtestservice/tests/testcase.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        return self.create_client_from_credential(
+            LoadTestClient,
+            credential=credential,
+            endpoint=endpoint,
+        )
+    
+    
+
+
+
```
Comment (msyyc) [inline sdk/loadtestservice/azure-analytics-loadtestservice/tests/testcase.py:32]: too many useless blank lines

**Comment by msyyc** [comment]
Hi @Yashikaj14 please update [ci.yaml](https://github.com/Azure/azure-sdk-for-python/blob/9002fb1f4ea9d0faeac8e6b475c4cf67bed34ad5/sdk/loadtestservice/ci.yml#L32-L33) with guidance: https://github.com/Azure/azure-sdk-for-python/wiki/Dataplane-Codegen-Quick-Start#createupdate-the-ciyaml

File: sdk/loadtestservice/azure-analytics-loadtestservice/README.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
+# Azure Analytics LoadTestService client library for Python
+<!-- write necessary description of service -->
```
Comment (msyyc) [inline sdk/loadtestservice/azure-analytics-loadtestservice/README.md:2]: This part is short description about your service, please update it

**Comment by msyyc** [comment]
CI failure of `core` is not related with this PR, please ignore it

File: sdk/loadtestservice/azure-analytics-loadtestservice/samples/sample.py [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+except KeyError:
+    LOG.error("Missing environment variable 'LOADTESTSERVICE_ENDPOINT' - please set if before running the example")
+    exit()
+
+# Build a client through AAD
+client = LoadTestClient(credential=DefaultAzureCredential(), endpoint=endpoint)
+
+# Listing the load test search
+try:
+    result = client.test.list_load_test_search()
```
Comment (msyyc) [inline sdk/loadtestservice/azure-analytics-loadtestservice/samples/sample.py:50]: Hi @Yashikaj14 the sample is not very common scenario for those customers who want to use loadtestservice, so please try to update it with more operation like `create, update, list and delete`

File: sdk/loadtestservice/azure-analytics-loadtestservice/tests/test_smoke.py [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+# license information.
+# -------------------------------------------------------------------------
+from testcase import LoadtestserviceTest, LoadtestservicePowerShellPreparer
+
+class LoadtestserviceSmokeTest(LoadtestserviceTest):
+
+    @LoadtestservicePowerShellPreparer()
+    def test_smoke_list_search(self, loadtestservice_endpoint):
+        client = self.create_client(loadtestservice_endpoint)
+        result = client.test.list_load_test_search()
```
Comment (msyyc) [inline sdk/loadtestservice/azure-analytics-loadtestservice/tests/test_smoke.py:14]: same with https://github.com/Azure/azure-sdk-for-python/pull/24744/files#r901233572

File: sdk/loadtestservice/azure-analytics-loadtestservice/tests/test_smoke_async.py [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+from testcase_async import LoadtestserviceAsyncTest
+
+
+class LoadtestserviceSmokeAsyncTest(LoadtestserviceAsyncTest):
+
+    @LoadtestservicePowerShellPreparer()
+    async def test_smoke_create_or_update_test(self, loadtestservice_endpoint):
+        client = self.create_client(loadtestservice_endpoint)
+        test_id="d7c9990e"  
+        body_test={"resourceId":"/subs","testId":"d7c9990e","description":"","displayName":"newtest","loadTestConfig":{"engineSize":"m","engineInstances":1,"splitAllCSVs":False},"secrets":{},"environmentVariables":{},"passFailCriteria":{"passFailMetrics":{}},"keyvaultReferenceIdentityType":"SystemAssigned","keyvaultReferenceIdentityId":None}
```
Comment (msyyc) [inline sdk/loadtestservice/azure-analytics-loadtestservice/tests/test_smoke_async.py:17]: @Yashikaj14 please format the code with IDE. It is hard to read if this line is too long

File: sdk/loadtestservice/azure-analytics-loadtestservice/samples/sample.py [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+    result=client.test_run.stop_test_run(test_run_id)
+except HttpResponseError as e:
+    print('Failed to send JSON message: {}'.format(e.response.json()))
+
+# Deleting a load test
+try:
+    test_id="test-name"
+    result=client.test.delete_load_test(test_id)
+except HttpResponseError as e:
+    print('Failed to send JSON message: {}'.format(e.response.json()))
```
Comment (msyyc) [inline sdk/loadtestservice/azure-analytics-loadtestservice/samples/sample.py:119]: same with https://github.com/Azure/azure-sdk-for-python/pull/24744/files#r902190298. @Yashikaj14

**Review by iscai-msft — APPROVED** [review]

File: sdk/loadtestservice/azure-analytics-loadtestservice/azure/analytics/loadtestservice/operations/_patch.py [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+class TestOperations(TestOperationsGenerated):
+    """
+    for performing the operations on test
+    """
+    def __init__(self, *args, **kwargs):
+        super(TestOperations, self).__init__(*args, **kwargs)
+
+    def upload_test_file(
+        self,
+        test_id,
```
[the remark this answers is not in the document]
Comment (Yashikaj14) [reply sdk/loadtestservice/azure-analytics-loadtestservice/azure/analytics/loadtestservice/operations/_patch.py]: @iscai-msft I have updated it.

**Review by msyyc — APPROVED** [review]

File: sdk/loadtestservice/azure-analytics-loadtestservice/azure/analytics/loadtestservice/operations/_patch.py [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+
+    def upload_test_file(
+        self,
+        test_id: str,
+        file_id: str,
+        file_content: JSON,
+        **kwargs
+    ) -> JSON:
+        """
+        Uploading a test file
```
Comment (iscai-msft) [inline sdk/loadtestservice/azure-analytics-loadtestservice/azure/analytics/loadtestservice/operations/_patch.py:76]: there should be documentation for each of the params, you can look at the generated code to see what we usually have for documentation

File: sdk/loadtestservice/azure-analytics-loadtestservice/samples/sample.py [hunk 14]
```diff
@@ -0,0 +1,119 @@
...
+from azure.identity import DefaultAzureCredential
+from azure.core.exceptions import HttpResponseError
+
+logging.basicConfig(level=logging.DEBUG)
+LOG = logging.getLogger()
+
+# Set the values of the client ID, tenant ID, and client secret of the AAD application as environment variables:
+# AZURE_CLIENT_ID, AZURE_TENANT_ID, AZURE_CLIENT_SECRET, LOADTESTSERVICE_ENDPOINT
+try:
+    endpoint = os.environ["LOADTESTSERVICE_ENDPOINT"]
```
Comment (iscai-msft) [inline sdk/loadtestservice/azure-analytics-loadtestservice/samples/sample.py:40]: nit: you can just rely on the KeyError here, i.e. you can just do `endpoint = os.environ["LOADTESTSERVICE_ENDPOINT"]` and an appropriate error will be thrown

File: sdk/loadtestservice/azure-analytics-loadtestservice/setup.py [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+    packages=find_packages(
+        exclude=[
+            # Exclude packages that will be covered by PEP420 or nspkg
+            "azure",
+            "azure.analytics",
+        ]
+    ),
+    install_requires=[
+        "azure-core<2.0.0,>=1.23.0",
+        "msrest>=0.6.21",
```
Comment (iscai-msft) [inline sdk/loadtestservice/azure-analytics-loadtestservice/setup.py:64]: do you actually need an msrest dependency? I don't think you do. I recommend you remove this dependency unless absolutely necessary. You will probably have to add an `isodate` dependency, you can look at the [changelog](https://github.com/Azure/autorest.python/blob/autorestv3/ChangeLog.md#2022-xx-xx---610) to see our mindep versions

File: sdk/loadtestservice/azure-analytics-loadtestservice/swagger/README.md [hunk 16]
```diff
@@ -0,0 +1,16 @@
...
+```yaml
+input-file: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/loadtestservice/data-plane/Microsoft.LoadTestService/preview/2022-06-01-preview/loadtestservice.json
+output-folder: ../azure/analytics/loadtestservice
+namespace: azure.analytics.loadtestservice
+package-name: azure-analytics-loadtestservice
+license-header: MICROSOFT_MIT_NO_VERSION
+clear-output-folder: true
+no-namespace-folders: true
+title: LoadTestClient
+version-tolerant: true
```
Comment (iscai-msft) [inline sdk/loadtestservice/azure-analytics-loadtestservice/swagger/README.md:12]: nit: no need for `version-tolerant: true` anymore bc we now default to version tolerant

File: sdk/loadtestservice/azure-analytics-loadtestservice/tests/test_smoke.py [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+        assert result is not None
+
+    @LoadtestservicePowerShellPreparer()
+    def test_smoke_list_search(self, loadtestservice_endpoint):
+        client = self.create_client(loadtestservice_endpoint)
+        result = client.test.list_load_test_search()
+        assert result is not None
+
+    @LoadtestservicePowerShellPreparer()
+    def test_smoke_delete_test(self, loadtestservice_endpoint):
```
Comment (iscai-msft) [inline sdk/loadtestservice/azure-analytics-loadtestservice/tests/test_smoke.py:30]: please add a test for `upload_file_test` bc that's a handwritten method

**Comment by msyyc** [comment]
https://github.com/Azure/azure-sdk-for-python/pull/24744#issuecomment-1186949389

File: .vscode/cspell.json [hunk 18]
```diff
@@ -447,6 +447,12 @@
         "undoc"
       ]
     },
+    {
+      "filename": "sdk/loadtestservice/azure-analytics-loadtestservice/azure/analytics/loadtestservice/**",
+      "words":[
+        "vusers"
+      ]
+    },
```
Comment (msyyc) [inline .vscode/cspell.json:455]: mark

File: sdk/loadtestservice/azure-analytics-loadtestservice/README.md [hunk 19]
```diff
@@ -0,0 +1,115 @@
...
+
+### Installating the package
+
+```bash
+python -m pip install azure-analytics-loadtestservice
+```
+
+#### Prequisites
+
+- Python 3.6 or later is required to use this package.
```
[the remark this answers is not in the document]
Comment (NiveditJain) [reply sdk/loadtestservice/azure-analytics-loadtestservice/README.md:19]: Resolved

File: sdk/loadtestservice/azure-analytics-loadtestservice/README.md [hunk 20]
```diff
@@ -0,0 +1,115 @@
...
+After setup, you can choose which type of [credential][azure_identity_credentials] from azure.identity to use.
+As an example, [DefaultAzureCredential][default_azure_credential] can be used to authenticate the client:
+
+Set the values of the client ID, tenant ID, and client secret of the AAD application as environment variables:
+`AZURE_CLIENT_ID`, `AZURE_TENANT_ID`, `AZURE_CLIENT_SECRET`
+
+Use the returned token credential to authenticate the client:
+
+```python
+>>> from azure.analytics.loadtestservice import LoadTestClient
```
[the remark this answers is not in the document]
Comment (NiveditJain) [reply sdk/loadtestservice/azure-analytics-loadtestservice/README.md:39]: Resolved

File: sdk/loadtestservice/azure-analytics-loadtestservice/tests/test_smoke.py (hunk 17 above)
Comment (NiveditJain) [reply sdk/loadtestservice/azure-analytics-loadtestservice/tests/test_smoke.py:30]: Duplicate at: https://github.com/Azure/azure-sdk-for-python/pull/25264#discussion_r925286925

File: sdk/loadtestservice/azure-analytics-loadtestservice/swagger/README.md (hunk 16 above)
Comment (NiveditJain) [reply sdk/loadtestservice/azure-analytics-loadtestservice/swagger/README.md:12]: Resolved

File: sdk/loadtestservice/azure-analytics-loadtestservice/setup.py (hunk 15 above)
Comment (NiveditJain) [reply sdk/loadtestservice/azure-analytics-loadtestservice/setup.py:64]: Resolved

File: sdk/loadtestservice/azure-analytics-loadtestservice/samples/sample.py (hunk 14 above)
Comment (NiveditJain) [reply sdk/loadtestservice/azure-analytics-loadtestservice/samples/sample.py:40]: Resloved

**Pull request closed without merging**